commands/publish_dafsa.py

```diff
@@ -0,0 +1,123 @@
+
+SERVER = "https://kinto.dev.mozaws.net/v1"
+CREDENTIALS = (os.getenv("USERNAME"), os.getenv("PASSWORD"))  # (username, password)
+
+BUCKET_ID = "firefox-core-network-dns"
+COLLECTION_ID = "public-suffix-list"
+RECORD_ID = "latest-commit-hash"
+
+
+def handle_request_errors(func):
+    def inner(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except HTTPError as http_err:
+            print(f"HTTP error occurred: {http_err}")
+        except Exception as err:
+            print(f"Other error occurred: {err}")
+        else:
+            print("Fetching Failed")
+
+    return inner
+
+
+@handle_request_errors
+def get_latest_hash():
+    response = requests.get(COMMIT_HASH_URL)
+    response.raise_for_status()
+
+    modified_files = response.json()["files"]
+    for f in modified_files:
+        if f["filename"] == "public_suffix_list.dat":
+            return response.json()["sha"]
+    return ""
+
+
+@handle_request_errors
+def download_resources(directory, *urls):
+    for url in urls:
+        # file_location is found by appending the file_name(at the end of url string) to temp directory
+        file_name = os.path.basename(url)
+        file_location = os.path.join(directory, file_name)
+        print(file_location)
+        response = requests.get(url, stream=True)
+        response.raise_for_status()
+
+        with open(file_location, "wb") as f:
+            for chunk in response.iter_content(chunk_size=1024):
+                if chunk:
+                    f.write(chunk)
+                else:
+                    print("Error!!!")
+
+
+def make_dafsa_and_publish(client, latest_hash):
+    with tempfile.TemporaryDirectory() as tmp:
+        download_resources(tmp, LIST_URL, MAKE_DAFSA_PY, PREPARE_TLDS_PY)
+        """
+        prepare_tlds.py is called with the two arguments the location of 
+        the downloaded public suffix list and the name of the output file
+        """
+        output_binary_name = "etld_data.json"
+
+        prepare_tlds_py_path = os.path.join(tmp, "prepare_tlds.py")
+        raw_psl_path = os.path.join(tmp, "public_suffix_list.dat")
+        output_binary_path = os.path.join(tmp, output_binary_name)
+
+        # Make the DAFSA
+        subprocess.run(
+            ["python3", prepare_tlds_py_path, raw_psl_path, output_binary_path]
+        )
+
+        subprocess.run(["ls", tmp])
+
+        # Upload the record
+        client.update_record(id=RECORD_ID, data={"latest-commit-hash": latest_hash})
+
+        # Upload the attachment
+        mimetype = "application/octet-stream"
+        filename = output_binary_name
+        filecontent = open(output_binary_path, "rb").read()
+        record_uri = client.get_endpoint("record", id=RECORD_ID)
+        attachment_uri = f"{record_uri}/attachment"
+        multipart = [("attachment", (filename, filecontent, mimetype))]
+        body, _ = client.session.request(
+            method="post", endpoint=attachment_uri, files=multipart
```

Comment:
I think you can pass `data={"latest-commit-hash"}` here and it will set both attributes and attachment in one operation, but I'm not sure ;)

Reply:
Also, maybe `latest-commit-hash` could just be called `commit` or `version` no?

Reply:
Let's split the difference, we'll call it `commit-hash` :)